autogpt_platform/backend/test/load_store_agents.py

```diff
@@ -0,0 +1,431 @@
+
+    # Determine if this should be approved
+    is_approved = metadata["is_available"]
+    submission_status = (
+        prisma.enums.SubmissionStatus.APPROVED
+        if is_approved
+        else prisma.enums.SubmissionStatus.PENDING
+    )
+
+    # Create the store listing first (without activeVersionId - will update after)
+    await db.storelisting.create(
+        data=StoreListingCreateInput(
+            id=listing_id,
+            slug=metadata["slug"],
+            agentGraphId=graph_id,
+            agentGraphVersion=graph_version,
+            owningUserId=AUTOGPT_USER_ID,
+            hasApprovedVersion=is_approved,
+            useForOnboarding=metadata["use_for_onboarding"],
+        )
+    )
+
+    # Create the store listing version
+    await db.storelistingversion.create(
+        data=StoreListingVersionCreateInput(
+            id=version_id,
+            version=1,
+            agentGraphId=graph_id,
+            agentGraphVersion=graph_version,
+            name=metadata["agent_name"],
+            subHeading=metadata["sub_heading"],
+            videoUrl=metadata["agent_video"],
+            imageUrls=metadata["agent_image"],
+            description=metadata["description"],
+            categories=metadata["categories"],
+            isFeatured=metadata["featured"],
+            isAvailable=metadata["is_available"],
+            submissionStatus=submission_status,
+            submittedAt=datetime.now() if is_approved else None,
+            reviewedAt=datetime.now() if is_approved else None,
+            storeListingId=listing_id,
+        )
+    )
+
+    # Update the store listing with the active version if approved
+    if is_approved:
+        await db.storelisting.update(
+            where={"id": listing_id},
+            data={"ActiveVersion": {"connect": {"id": version_id}}},
+        )
+
+
+async def main():
+    """Main function to load all store agents."""
+    print("=" * 60)
+    print("Loading Store Agents into Test Database")
+    print("=" * 60)
+
+    db = Prisma()
+    await db.connect()
+
+    try:
+        # Step 0: Initialize agent blocks
+        print("\n[Step 0] Initializing agent blocks...")
+        known_blocks = await initialize_blocks(db)
+
+        # Step 1: Create user and profile
+        print("\n[Step 1] Creating user and profile...")
+        await create_user_and_profile(db)
+
+        # Step 2: Load CSV metadata
+        print("\n[Step 2] Loading CSV metadata...")
+        csv_metadata = await load_csv_metadata()
+        print(f"  Found {len(csv_metadata)} store listing entries in CSV")
+
+        # Step 3: Find all JSON files and match with CSV
+        print("\n[Step 3] Processing agent JSON files...")
+        json_files = list(AGENTS_DIR.glob("agent_*.json"))
+        print(f"  Found {len(json_files)} agent JSON files")
+
+        # Build mapping from version_id to json file
+        loaded_graphs = {}  # graph_id -> (graph_id, version)
+
+        for json_file in json_files:
```

Comment:
you need to do it with a transaction, whether the whole flow or per agent, otherwise you'll leave with dangling resource